components/eventForm/LocationBanner.tsx

```diff
@@ -0,0 +1,102 @@
+import { useReverseGeocodeQuery } from "../../hooks/Geocoding"
+import { placemarkToFormattedAddress } from "../../lib/location"
+import React from "react"
+import { StyleSheet, View } from "react-native"
+import {
+  useEventFormContext
+} from "./EventForm"
+import { FormLabel, SkeletonFormLabel } from "../formComponents/FormLabels"
+import { MaterialIcons } from "@expo/vector-icons"
+import { FontScaleFactors, useFontScale } from "../../lib/FontScale"
+import { EventFormLocationInfo, EventFormPlacemarkInfo } from "./EventFormValues"
+
+/**
+ * Displays the selected location (if one) in the event form.
+ */
+export const EventFormLocationBanner = () => {
+  const locationInfo = useEventFormContext().watch("locationInfo")
+  return (
+    <View style={styles.bannerContainer}>
+      {!locationInfo
+        ? (
+          <FormLabel
+            style={styles.label}
+            icon="location-pin"
+            headerText="No Location Selected"
+            captionText="You must select a location to save this event."
+          />
+        )
+        : (
+          <LocationInfoBanner {...locationInfo} />
+        )}
+      <MaterialIcons
+        name="chevron-right"
+        size={
+          24 * useFontScale({ maximumScaleFactor: FontScaleFactors.xxxLarge })
+        }
+        color="black"
+        style={styles.bannerNavigationIcon}
+      />
+    </View>
+  )
+}
+
+const LocationInfoBanner = (locationInfo: EventFormLocationInfo) => {
+  return locationInfo.placemarkInfo
+    ? (
+      <PlacemarkInfoBanner {...locationInfo.placemarkInfo} />
```

Comment:
To make sure I understand this correctly, the first state is if there's already an address saved in the event and the second state is if there isn't and we need to reverse geocode the address?

Reply:
Yes